x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/agent_config/create_datasource_page/step_review.tsx

```diff
@@ -3,19 +3,162 @@
+
+  // Config information
+  const {
+    fleet: { enabled: isFleetEnabled },
+  } = useConfig();
+
+  // Retrieve agent count
+  useEffect(() => {
+    const getAgentCount = async () => {
+      const { data } = await sendGetAgentStatus({ configId: agentConfig.id });
+      if (data?.results.total) {
+        setAgentCount(data.results.total);
+      }
+    };
+
+    if (isFleetEnabled) {
+      getAgentCount();
+    }
+  }, [agentConfig.id, isFleetEnabled]);
+
+  const showAgentDisclaimer = isFleetEnabled && agentCount;
+  const fullAgentDatasource = storedDatasourceToAgentDatasource(datasource);
+
   return (
-    <EuiFlexGroup direction="column" gutterSize="none">
+    <EuiFlexGroup direction="column">
+      <EuiFlexItem>
+        <EuiFlexGroup justifyContent="spaceBetween">
+          <EuiFlexItem grow={false}>
+            <EuiTitle size="s">
+              <h3>
+                <FormattedMessage
+                  id="xpack.ingestManager.createDatasource.stepReview.reviewTitle"
+                  defaultMessage="Review changes"
+                />
+              </h3>
+            </EuiTitle>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>{backLink}</EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiFlexItem>
+
+      {/* Agents affected warning */}
+      {showAgentDisclaimer ? (
+        <EuiFlexItem>
+          <EuiCallOut
+            title={
+              <FormattedMessage
+                id="xpack.ingestManager.createDatasource.stepReview.agentsAffectedCalloutTitle"
+                defaultMessage="This action will affect {count, plural, one {# agent} other {# agents}}"
+                values={{
+                  count: agentCount,
+                }}
+              />
+            }
+          >
+            <EuiText>
+              <p>
+                <FormattedMessage
+                  id="xpack.ingestManager.createDatasource.stepReview.agentsAffectedCalloutText"
+                  defaultMessage="Fleet has detected that the selected agent configuration, {configName} is already in use by some your agents. As a result of this action, Fleet will update all agents that are enrolled with this configuration."
+                  values={{
+                    configName: <strong>{agentConfig.name}</strong>,
+                  }}
+                />
+              </p>
+            </EuiText>
+          </EuiCallOut>
+        </EuiFlexItem>
+      ) : null}
+
+      {/* Preview and YAML view */}
+      {/* TODO: Implement preview tab */}
+      <EuiFlexItem>
+        <EuiTabbedContent
+          tabs={[
+            {
+              id: 'yaml',
+              name: i18n.translate('xpack.ingestManager.agentConfigInfo.yamlTabName', {
+                defaultMessage: 'YAML',
+              }),
+              content: (
+                <Fragment>
+                  <EuiSpacer size="s" />
+                  <EuiCodeBlock language="yaml" isCopyable overflowHeight={450}>
```

Comment:
I assume something similar to this will be used in the `yaml` tab of the Agent config 😃

Reply:
yep!